Classes/TestableFramework.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/*
+ * (c) Contributors of the Neos Project - www.neos.io
+ * Please see the LICENSE file which was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Neos\BuildEssentials;
+
+use Neos\Flow\Annotations as Flow;
+use Neos\Flow\Configuration\ConfigurationManager;
+use Neos\Flow\Core\Bootstrap;
+use Neos\Flow\Core\RequestHandlerInterface;
+use Neos\Flow\Persistence\PersistenceManagerInterface;
+use Neos\Utility\Files;
+
+/**
+ * Low level operations for use in testing, all to be used in static calls.
+ *
+ * @Flow\Proxy(false)
+ */
+abstract readonly class TestableFramework
+{
+    /**
+     * Calculates a FLOW_ROOTPATH
+     * @return string
+     */
+    public static function calculateRootPath(): string
+    {
+        $flowRootPathFromEnvironment = trim(getenv('FLOW_ROOTPATH') ?: '', '"\' ');
+        return $flowRootPathFromEnvironment ?: dirname(__DIR__, 3) . '/';
+    }
+
+    public static function getApplicationContext(): string
+    {
+        $flowContextFromEnvironment = trim(getenv('FLOW_CONTEXT') ?: '', '"\' ');
+        return $flowContextFromEnvironment ?: 'Testing';
+    }
+
+    /**
+     * Build a Flow Bootstrap with an instance of the given class implementing RequestHandlerInterface
+     *
+     * @param string $context
+     * @param class-string<RequestHandlerInterface> $requestHandlerClassname
+     * @return Bootstrap
+     */
+    public static function buildBootstrapWithPreselectedRequestHandler(string $context, string $requestHandlerClassname): Bootstrap
```

Comment:
hmm the method provided me good guidance how to boot flow but i wanted to manipulate the state of the requesthandler beforehand and set the http request. 

otherwise i would need to do

```php
/** @var RuntimeSequenceHttpRequestHandler $requestHandler */
$requestHandler = self::$bootstrap->getActiveRequestHandler();
```

We could ofcourse allow an instance here ... but then again it shows some inflexibility and thinking about this it should actually probably be part of the flow `Bootstrap` itself?

because otherwise i would also need another require for 

```
require_once dirname(__DIR__, 4) . '/Build/BuildEssentials/Classes/TestableFramework.php';
```